lib/node_modules/@stdlib/number/float64/base/normalize/include/stdlib/number/float64/base/normalize.h

```diff
@@ -0,0 +1,65 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifndef STDLIB_NUMBER_FLOAT64_BASE_NORMALIZE_H
+#define STDLIB_NUMBER_FLOAT64_BASE_NORMALIZE_H
+
+#include <stdint.h>
+
+/*
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Returns a normal number y and exponent exp satisfying x = y * 2^exp.
+*
+* @example
+* #include <stdint.h>
+*
+* stdlib_base_float64_normalizer_t normalizer;
+*
+* // Assign a double precision floating-point number:
+* normalizer.value = 3.14;
+*
+* // Extract the exponent:
+* int32_t exp = normalizer.parts.exp;
+* // Extract the number:
+* double y = normalizer.parts.y;
+*/
+typedef union{
+	double value;
+	struct{
+		int32_t exp;
+		double y;
+	} parts;
+} stdlib_base_float64_normalizer_t;
+
+
+/**
+* Returns a normal number y and exponent exp satisfying x = y * 2^exp.
+*/
+void stdlib_base_float64_normalize( const double x );
```

Comment:
Instead of working with a union, I'd do

```c
void stdlib_base_float64_normalize( const double x, double *y, int32_t *exp );
```

[`@stdlib/number/float64/base/to-words`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/number/float64/base/to-words#c-apis) provides an example of such an API.

Reply:
Ohh okay, that's great, now it's much easier to understand, thank you atan! 🥳

Reply:
I will try to complete this PR by tommorow ( my time )